theano/gpuarray/dnn_fwd.c

```diff
@@ -142,6 +143,9 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
           return 1;
         }
         algo = choice.algo;
+        params->conv_algo = algo;
+        params->conv_ws_size = choice.memory;
+        params->conv_tensor_op = choice.mathType;
```

Comment:
I don't think this is a good use of params. Can you just create new variable like algo (around line 86) instead for those 3 extra information?

Why did you needed to modify params?